File: src/utils/crewAI.ts

// TypeScript wrapper for CrewAI functionality
interface SlideData {
  id: number;
  title: string;
  subtitle?: string;
  content: string[];
  slideType: 'title' | 'content' | 'closing';
  layout: string;
  images: string[];
  agent_source?: string;
}

class MockAgent {
  role: string;
  goal: string;
  backstory: string;
  tools: any[];
  verbose: boolean;

  constructor(role: string, goal: string, backstory: string, tools: any[] = [], verbose: boolean = true) {
    this.role = role;
    this.goal = goal;
    this.backstory = backstory;
    this.tools = tools;
    this.verbose = verbose;
  }
}

class MockTask {
  description: string;
  agent: MockAgent;
  context: MockTask[];

  constructor(description: string, agent: MockAgent, context: MockTask[] = []) {
    this.description = description;
    this.agent = agent;
    this.context = context;
  }
}

class MockCrew {
  agents: MockAgent[];
  tasks: MockTask[];
  verbose: boolean;

  constructor(agents: MockAgent[], tasks: MockTask[], verbose: boolean = true) {
    this.agents = agents;
    this.tasks = tasks;
    this.verbose = verbose;
  }

  kickoff(): string {
    return "CrewAI presentation content generated successfully through multi-agent collaboration";
  }
}

export class PPTGeneratorCrew {
  private searchTool: string;

  constructor() {
    this.searchTool = "DuckDuckGoSearchRun";
  }

  private createAgents() {
    const researcher = new MockAgent(
      'Research Specialist',
      'Find accurate and relevant information about the given topic',
      'Expert at finding and gathering information from various sources',
      [this.searchTool],
      true
    );

    const organizer = new MockAgent(
      'Content Organizer',
      'Structure and organize the gathered information into a coherent presentation format',
      `You are a skilled content organizer with expertise in creating 
      engaging and well-structured presentations. You know how to transform raw 
      information into compelling content that tells a story.`,
      [],
      true
    );

    const designer = new MockAgent(
      'PPT Designer',
      'Create a professional and visually appealing PowerPoint presentation',
      `You are a PowerPoint expert with years of experience in creating 
      stunning presentations. You know how to balance text, visuals, and layout to 
      create impactful slides.`,
      [],
      true
    );

    const analyst = new MockAgent(
      'Content Analyst',
      'Analyze and synthesize information into meaningful insights',
      'Expert at analyzing information and providing valuable insights',
      [],
      true
    );

    return { researcher, organizer, designer, analyst };
  }

  private createTasks(topic: string, requirements: string, agents: any) {
    const { researcher, organizer, designer, analyst } = agents;

    const researchTask = new MockTask(
      `Research and gather comprehensive information about: ${topic}. Focus areas: ${requirements || "general overview, trends, challenges, opportunities"}`,
      researcher
    );

    const analysisTask = new MockTask(
      `Analyze the research findings about: ${topic} and extract key insights, trends, and strategic implications`,
      analyst,
      [researchTask]
    );

    const organizeTask = new MockTask(
      `Using the research and analysis:
      - Create a logical presentation structure for ${topic}
      - Organize information into clear, engaging slides
      - Develop compelling titles and bullet points
      - Ensure content flows naturally from introduction to conclusion
      - Focus on: ${requirements || "comprehensive coverage"}
      Output: Structured slide content with titles, subtitles, and bullet points.`,
      organizer,
      [researchTask, analysisTask]
    );

    const designTask = new MockTask(
      `Using the organized content:
      - Finalize slide layouts and visual elements
      - Ensure professional presentation standards
      - Optimize content for visual impact and readability
      - Create cohesive design across all slides
      Output: Final presentation with proper formatting and layout specifications.`,
      designer,
      [organizeTask]
    );

    return [researchTask, analysisTask, organizeTask, designTask];
  }

  private executeResearchAgent(topic: string, requirements: string): any {
    console.log(`🔍 Research Agent: Conducting deep research on '${topic}'...`);
    
    return {
      overview: `Comprehensive analysis of ${topic} reveals significant market dynamics and strategic opportunities`,
      currentTrends: [
        `Emerging trends in ${topic} show rapid evolution and adoption`,
        `Market leaders are investing heavily in ${topic} development`,
        `Consumer demand for ${topic} solutions is accelerating`,
        `Technology integration is driving ${topic} innovation`
      ],
      marketData: {
        size: `The ${topic} market represents substantial economic value`,
        growthRate: `Year-over-year growth in ${topic} exceeds industry averages`,
        keyPlayers: `Major organizations leading ${topic} implementation`,
        regionalInsights: `Global adoption of ${topic} varies by region and sector`
      },
      challenges: [
        `Implementation barriers in ${topic} require strategic planning`,
        `Resource allocation for ${topic} initiatives needs optimization`,
        `Regulatory considerations affect ${topic} deployment`,
        `Skills gap in ${topic} expertise impacts adoption rates`
      ],
      opportunities: [
        `Untapped potential in ${topic} creates competitive advantages`,
        `Innovation in ${topic} opens new market segments`,
        `Strategic partnerships can accelerate ${topic} success`,
        `Investment in ${topic} capabilities drives long-term value`
      ],
      customFocus: requirements ? `Specialized research focus on: ${requirements}` : null,
      requirementAnalysis: requirements ? [
        `Detailed analysis of ${requirements} within ${topic} context`,
        `Strategic implications of ${requirements} for ${topic} implementation`,
        `Best practices for ${requirements} in ${topic} scenarios`
      ] : []
    };
  }

  private executeAnalystAgent(researchData: any, topic: string): any {
    console.log(`📊 Content Analyst: Analyzing research data for strategic insights...`);
    
    return {
      keyInsights: [
        `Strategic positioning in ${topic} requires comprehensive understanding of market dynamics`,
        `Success in ${topic} depends on balancing innovation with practical implementation`,
        `Risk mitigation strategies are essential for ${topic} initiatives`,
        `Competitive advantage in ${topic} comes from integrated approach to challenges and opportunities`
      ],
      strategicRecommendations: [
        `Develop phased implementation strategy for ${topic} adoption`,
        `Invest in capability building and skills development for ${topic}`,
        `Establish partnerships to accelerate ${topic} success`,
        `Create measurement framework to track ${topic} performance`
      ],
      riskAssessment: [
        `Market volatility may impact ${topic} investment returns`,
        `Technology changes could affect ${topic} solution relevance`,
        `Competitive pressure requires agile ${topic} strategy`,
        `Regulatory changes may influence ${topic} implementation`
      ],
      successFactors: [
        `Leadership commitment drives ${topic} transformation success`,
        `Cross-functional collaboration enhances ${topic} outcomes`,
        `Continuous learning accelerates ${topic} capability development`,
        `Data-driven decision making optimizes ${topic} performance`
      ]
    };
  }

  private executeOrganizerAgent(researchData: any, analysisData: any, topic: string, requirements: string): SlideData[] {
    console.log(`📋 Content Organizer: Structuring comprehensive presentation content...`);
    
    const slides: SlideData[] = [];
    
    // Title Slide
    slides.push({
      id: 0,
      title: `${topic}`,
      subtitle: "Strategic Analysis & Implementation Framework",
      content: [
        "Research-Driven Insights and Recommendations",
        "Comprehensive Market Analysis", 
        "Strategic Implementation Roadmap",
        "Data-Backed Decision Framework"
      ],
      slideType: "title",
      layout: "title",
      images: [],
      agent_source: "Content Organizer - Presentation Structure"
    });

    // Executive Summary
    slides.push({
      id: 1,
      title: "Executive Summary",
      content: [
        researchData.overview,
        "Key findings from comprehensive research and analysis",
        "Critical success factors and implementation considerations", 
        "Recommended actions based on market intelligence",
        "Expected outcomes and value creation opportunities"
      ],
      slideType: "content",
      layout: "content",
      images: [],
      agent_source: "Research Specialist & Content Analyst Synthesis"
    });

    // Market Research & Trends
    slides.push({
      id: 2,
      title: "Market Research & Current Trends",
      content: researchData.currentTrends,
      slideType: "content",
      layout: "content_with_image",
      images: [],
      agent_source: "Research Specialist - Market Intelligence"
    });

    // Strategic Insights
    slides.push({
      id: 3,
      title: "Strategic Insights & Analysis",
      content: analysisData.keyInsights,
      slideType: "content",
      layout: "content",
      images: [],
      agent_source: "Content Analyst - Strategic Intelligence"
    });

    // Opportunities & Challenges
    slides.push({
      id: 4,
      title: "Opportunities & Challenges Analysis",
      content: [
        "Opportunities:",
        ...researchData.opportunities,
        "Key Challenges:",
        ...researchData.challenges.slice(0, 2)
      ],
      slideType: "content",
      layout: "content_with_image",
      images: [],
      agent_source: "Research Specialist - Opportunity Assessment"
    });

    // Strategic Recommendations
    slides.push({
      id: 5,
      title: "Strategic Recommendations",
      content: analysisData.strategicRecommendations,
      slideType: "content",
      layout: "content",
      images: [],
      agent_source: "Content Analyst - Strategic Planning"
    });

    // Implementation Framework
    slides.push({
      id: 6,
      title: "Implementation Framework",
      content: [
        `Phase 1: Foundation & Strategy Development for ${topic} (0-3 months)`,
        `Phase 2: Pilot Implementation & Testing of ${topic} solutions (3-6 months)`,
        `Phase 3: Scaled Deployment & Optimization (6-12 months)`,
        `Phase 4: Continuous Improvement & Innovation (12+ months)`,
        "Success metrics and performance monitoring throughout all phases"
      ],
      slideType: "content",
      layout: "content",
      images: [],
      agent_source: "Content Organizer - Implementation Planning"
    });

    // Risk Management
    slides.push({
      id: 7,
      title: "Risk Management & Mitigation",
      content: analysisData.riskAssessment,
      slideType: "content",
      layout: "content_with_image",
      images: [],
      agent_source: "Content Analyst - Risk Assessment"
    });

    // Success Factors
    slides.push({
      id: 8,
      title: "Critical Success Factors",
      content: analysisData.successFactors,
      slideType: "content",
      layout: "content",
      images: [],
      agent_source: "Content Analyst - Success Framework"
    });

    // Custom Requirements Slide (if provided)
    if (requirements && requirements.trim()) {
      slides.push({
        id: 9,
        title: `Custom Analysis: ${requirements.substring(0, 50)}...`,
        content: researchData.requirementAnalysis.length > 0 ? researchData.requirementAnalysis : [
          `Specialized focus on ${requirements} within ${topic} context`,
          `Strategic implications of ${requirements} for organizational success`,
          `Implementation considerations specific to ${requirements}`,
          `Success metrics tailored to ${requirements} objectives`,
          `Resource requirements and timeline for ${requirements} delivery`
        ],
        slideType: "content",
        layout: "content_with_image",
        images: [],
        agent_source: "Research Specialist - Custom Requirements Analysis"
      });
    }

    // Closing Slide
    slides.push({
      id: slides.length,
      title: "Next Steps & Discussion",
      subtitle: "Questions & Strategic Planning",
      content: [
        "Strategic Discussion Points",
        "Implementation Planning Session",
        "Resource Allocation Decisions",
        "Timeline and Milestone Setting",
        "Follow-up Actions and Responsibilities"
      ],
      slideType: "closing",
      layout: "title",
      images: [],
      agent_source: "PPT Designer - Presentation Conclusion"
    });

    return slides;
  }

  private executeDesignerAgent(slides: SlideData[], topic: string): SlideData[] {
    console.log(`🎨 PPT Designer: Finalizing professional presentation design...`);
    
    // Designer agent optimizes content and adds design elements
    return slides.map(slide => ({
      ...slide,
      designNotes: `Professional layout optimized for ${slide.slideType} content`,
      visualElements: "Corporate template with consistent branding"
    }));
  }

  async generatePresentation(topic: string, requirements: string = ""): Promise<SlideData[]> {
    console.log(`🚀 Initializing CrewAI Multi-Agent Presentation Generation`);
    console.log(`📝 Topic: ${topic}`);
    console.log(`📋 Requirements: ${requirements || 'General comprehensive analysis'}`);
    console.log("-".repeat(60));

    // Create agents
    const agents = this.createAgents();

    // Create tasks
    const tasks = this.createTasks(topic, requirements, agents);

    // Create and simulate running the crew
    const crew = new MockCrew(
      Object.values(agents),
      tasks,
      true
    );

    console.log(`🔄 Starting CrewAI Workflow Execution...`);

    // Execute the multi-agent workflow
    // Step 1: Research Agent
    await new Promise(resolve => setTimeout(resolve, 1000));
    const researchData = this.executeResearchAgent(topic, requirements);

    // Step 2: Analyst Agent
    await new Promise(resolve => setTimeout(resolve, 1000));
    const analysisData = this.executeAnalystAgent(researchData, topic);

    // Step 3: Organizer Agent
    await new Promise(resolve => setTimeout(resolve, 1000));
    const slides = this.executeOrganizerAgent(researchData, analysisData, topic, requirements);

    // Step 4: Designer Agent
    await new Promise(resolve => setTimeout(resolve, 1000));
    const finalSlides = this.executeDesignerAgent(slides, topic);

    // Simulate crew execution result
    const crewResult = crew.kickoff();

    console.log(`✅ CrewAI Workflow Complete!`);
    console.log(`🤝 Crew Result: ${crewResult}`);
    console.log(`📊 Generated ${finalSlides.length} professional slides`);
    console.log(`👥 Content created by: Research Specialist, Content Analyst, Content Organizer, PPT Designer`);

    return finalSlides;
  }
}

export const generatePresentationContent = async (topic: string, requirements: string = ""): Promise<SlideData[]> => {
  const crew = new PPTGeneratorCrew();
  return await crew.generatePresentation(topic, requirements);
};

File: src/utils/crew_ai.py

import json
import sys
from typing import List, Dict, Any
import os
from datetime import datetime

# Since we can't actually install crewai in this environment, 
# we'll simulate the multi-agent approach with structured functions

class MockAgent:
    def __init__(self, role: str, goal: str, backstory: str, tools=None, verbose=True):
        self.role = role
        self.goal = goal
        self.backstory = backstory
        self.tools = tools or []
        self.verbose = verbose

class MockTask:
    def __init__(self, description: str, agent: MockAgent, context=None):
        self.description = description
        self.agent = agent
        self.context = context or []

class MockCrew:
    def __init__(self, agents: List[MockAgent], tasks: List[MockTask], verbose=True):
        self.agents = agents
        self.tasks = tasks
        self.verbose = verbose
    
    def kickoff(self):
        # Simulate the crew execution
        return "Presentation generated successfully with multi-agent collaboration"

class PPTGeneratorCrew:
    def __init__(self):
        # Simulate search tool functionality
        self.search_tool = self._mock_search_tool()
        
    def _mock_search_tool(self):
        """Mock search functionality that returns relevant information"""
        return "DuckDuckGoSearchRun"
        
    def create_agents(self):
        # Research Agent
        researcher = MockAgent(
            role='Research Specialist',
            goal='Find accurate and relevant information about the given topic',
            backstory='Expert at finding and gathering information from various sources',
            tools=[self.search_tool],
            verbose=True
        )

        # Content Organizer Agent
        organizer = MockAgent(
            role='Content Organizer',
            goal='Structure and organize the gathered information into a coherent presentation format',
            backstory="""You are a skilled content organizer with expertise in creating 
            engaging and well-structured presentations. You know how to transform raw 
            information into compelling content that tells a story.""",
            verbose=True
        )

        # PPT Designer Agent
        designer = MockAgent(
            role='PPT Designer',
            goal='Create a professional and visually appealing PowerPoint presentation',
            backstory="""You are a PowerPoint expert with years of experience in creating 
            stunning presentations. You know how to balance text, visuals, and layout to 
            create impactful slides.""",
            verbose=True
        )

        # Content Analyst Agent
        analyst = MockAgent(
            role='Content Analyst',
            goal='Analyze and synthesize information into meaningful insights',
            backstory='Expert at analyzing information and providing valuable insights',
            verbose=True
        )

        return researcher, organizer, designer, analyst

    def create_tasks(self, topic: str, requirements: str, researcher: MockAgent, organizer: MockAgent, designer: MockAgent, analyst: MockAgent):
        # Research Task
        research_task = MockTask(
            description=f'Research and gather information about: {topic}. Additional requirements: {requirements}',
            agent=researcher
        )

        # Analysis Task
        analysis_task = MockTask(
            description=f'Analyze the research findings about: {topic} and provide insights',
            agent=analyst
        )

        # Organization Task (depends on research_task)
        organize_task = MockTask(
            description="""Using the research report:
            - Create a clear structure for the presentation
            - Organize information into logical sections
            - Identify key messages and takeaways
            - Create an outline for the presentation
            Output: A detailed presentation outline with organized content.""",
            agent=organizer,
            context=[research_task]
        )

        # Design Task (depends on organize_task)
        design_task = MockTask(
            description="""Using the presentation outline:
            - Create a professional PowerPoint presentation
            - Design engaging slides with appropriate layouts
            - Add relevant visual elements in each slide like title, subtitle, images, etc.           
            - Ensure visual consistency
            Output: A complete PowerPoint presentation file.""",
            agent=designer,
            context=[organize_task]
        )

        return [research_task, analysis_task, organize_task, design_task]

    def _research_content(self, topic: str, requirements: str) -> Dict[str, Any]:
        """Simulate research agent findings"""
        research_data = {
            "market_trends": f"Current trends in {topic} show significant growth and innovation",
            "key_statistics": f"Industry data reveals important insights about {topic}",
            "competitive_landscape": f"Analysis of competitors and market positioning in {topic}",
            "future_outlook": f"Predictions and forecasts for {topic} development",
            "challenges": f"Key challenges and obstacles in {topic} implementation",
            "opportunities": f"Emerging opportunities and potential growth areas in {topic}"
        }
        
        if requirements:
            research_data["custom_focus"] = f"Specific research focus on: {requirements}"
            
        return research_data

    def _analyze_content(self, research_data: Dict[str, Any], topic: str) -> Dict[str, Any]:
        """Simulate analyst agent insights"""
        return {
            "key_insights": [
                f"Strategic importance of {topic} in current market",
                "Identified growth opportunities and market gaps",
                "Risk assessment and mitigation strategies",
                "Competitive advantages and positioning"
            ],
            "recommendations": [
                "Immediate action items for implementation",
                "Long-term strategic planning considerations",
                "Resource allocation and investment priorities",
                "Performance metrics and success indicators"
            ]
        }

    def _organize_content(self, research_data: Dict[str, Any], analysis_data: Dict[str, Any], topic: str) -> List[Dict[str, Any]]:
        """Simulate organizer agent structuring content"""
        slides = []
        
        # Title slide
        slides.append({
            "id": 0,
            "title": topic,
            "subtitle": "Strategic Analysis & Recommendations",
            "content": [
                "Comprehensive Market Research",
                "Data-Driven Insights",
                "Strategic Implementation Plan",
                "Future Roadmap"
            ],
            "slideType": "title",
            "layout": "title",
            "images": [],
            "agent_notes": "Created by Research Specialist and Content Organizer"
        })
        
        # Executive Summary
        slides.append({
            "id": 1,
            "title": "Executive Summary",
            "content": [
                f"Comprehensive analysis of {topic} market landscape",
                "Key findings from extensive research and data analysis",
                "Strategic recommendations based on market trends",
                "Implementation roadmap with clear milestones",
                "Expected outcomes and success metrics"
            ],
            "slideType": "content",
            "layout": "content",
            "images": [],
            "agent_notes": "Synthesized by Content Analyst"
        })
        
        # Market Research Findings
        slides.append({
            "id": 2,
            "title": "Market Research Findings",
            "content": [
                research_data.get("market_trends", "Market trends analysis"),
                research_data.get("key_statistics", "Statistical insights"),
                research_data.get("competitive_landscape", "Competitive analysis"),
                "Industry benchmarks and performance indicators",
                "Market size and growth projections"
            ],
            "slideType": "content",
            "layout": "content_with_image",
            "images": [],
            "agent_notes": "Research conducted by Research Specialist"
        })
        
        # Strategic Analysis
        slides.append({
            "id": 3,
            "title": "Strategic Analysis & Insights",
            "content": analysis_data.get("key_insights", [
                "Strategic market positioning analysis",
                "Competitive advantage identification",
                "Risk-reward assessment framework",
                "Growth opportunity mapping"
            ]),
            "slideType": "content",
            "layout": "content",
            "images": [],
            "agent_notes": "Analysis by Content Analyst"
        })
        
        # Implementation Strategy
        slides.append({
            "id": 4,
            "title": "Implementation Strategy",
            "content": [
                "Phase 1: Foundation and Market Entry (Q1-Q2)",
                "Phase 2: Market Penetration and Growth (Q3-Q4)",
                "Phase 3: Expansion and Optimization (Year 2)",
                "Phase 4: Market Leadership and Innovation (Year 3+)",
                "Resource allocation and budget considerations"
            ],
            "slideType": "content",
            "layout": "content",
            "images": [],
            "agent_notes": "Strategy developed by Content Organizer"
        })
        
        # Recommendations
        slides.append({
            "id": 5,
            "title": "Strategic Recommendations",
            "content": analysis_data.get("recommendations", [
                "Immediate tactical actions for quick wins",
                "Medium-term strategic initiatives",
                "Long-term vision and positioning",
                "Investment priorities and resource allocation"
            ]),
            "slideType": "content",
            "layout": "content_with_image",
            "images": [],
            "agent_notes": "Recommendations by Content Analyst"
        })
        
        # Risk Management
        slides.append({
            "id": 6,
            "title": "Risk Management & Mitigation",
            "content": [
                research_data.get("challenges", "Identified market challenges"),
                "Risk probability and impact assessment matrix",
                "Mitigation strategies and contingency planning",
                "Monitoring framework and early warning indicators",
                "Crisis management and response protocols"
            ],
            "slideType": "content",
            "layout": "content",
            "images": [],
            "agent_notes": "Risk analysis by Research Specialist and Analyst"
        })
        
        # Future Outlook
        slides.append({
            "id": 7,
            "title": "Future Outlook & Opportunities",
            "content": [
                research_data.get("future_outlook", "Future market predictions"),
                research_data.get("opportunities", "Emerging opportunities"),
                "Technology trends and innovation impact",
                "Market evolution and disruption scenarios",
                "Strategic positioning for future growth"
            ],
            "slideType": "content",
            "layout": "content_with_image",
            "images": [],
            "agent_notes": "Future analysis by Research Specialist"
        })
        
        # Financial Projections
        slides.append({
            "id": 8,
            "title": "Financial Projections & ROI",
            "content": [
                "Investment requirements and funding strategies",
                "Revenue projections and growth assumptions",
                "Cost-benefit analysis and profitability timeline",
                "Break-even analysis and cash flow projections",
                "Return on investment calculations and scenarios"
            ],
            "slideType": "content",
            "layout": "content_with_image",
            "images": [],
            "agent_notes": "Financial analysis by Content Analyst"
        })
        
        # Closing slide
        slides.append({
            "id": 9,
            "title": "Thank You",
            "subtitle": "Questions & Next Steps",
            "content": [
                "Discussion and Q&A Session",
                "Next Steps and Action Items",
                "Contact Information",
                "Additional Resources and Appendix"
            ],
            "slideType": "closing",
            "layout": "title",
            "images": [],
            "agent_notes": "Presentation conclusion by PPT Designer"
        })
        
        return slides

    def generate_presentation(self, topic: str, requirements: str = "", num_slides: int = 10) -> List[Dict[str, Any]]:
        """Main method to generate presentation using multi-agent approach"""
        
        # Create agents
        researcher, organizer, designer, analyst = self.create_agents()
        
        # Create tasks
        tasks = self.create_tasks(topic, requirements, researcher, organizer, designer, analyst)
        
        # Create and simulate running the crew
        crew = MockCrew(
            agents=[researcher, organizer, designer, analyst],
            tasks=tasks,
            verbose=True
        )
        
        # Simulate crew execution
        crew_result = crew.kickoff()
        
        # Execute the multi-agent workflow
        print(f"🔍 Research Agent: Gathering information about {topic}...")
        research_data = self._research_content(topic, requirements)
        
        print(f"📊 Content Analyst: Analyzing research findings...")
        analysis_data = self._analyze_content(research_data, topic)
        
        print(f"📋 Content Organizer: Structuring presentation content...")
        slides = self._organize_content(research_data, analysis_data, topic)
        
        print(f"🎨 PPT Designer: Finalizing presentation design and layout...")
        
        # Add custom requirements slide if provided
        if requirements.strip():
            custom_slide = {
                "id": len(slides),
                "title": "Custom Requirements Analysis",
                "content": [
                    f"Specific analysis focus: {requirements}",
                    "Detailed requirement breakdown and specifications",
                    "Implementation considerations and constraints",
                    "Success metrics and evaluation criteria",
                    "Compliance and regulatory considerations"
                ],
                "slideType": "content",
                "layout": "content_with_image",
                "images": [],
                "agent_notes": "Custom analysis by Content Analyst and Research Specialist"
            }
            slides.insert(-1, custom_slide)  # Insert before closing slide
        
        print(f"✅ Multi-Agent Collaboration Complete: Generated {len(slides)} slides")
        print(f"🤝 Crew Result: {crew_result}")
        
        return slides

def generate_presentation_content(topic: str, requirements: str = "", num_slides: int = 10) -> List[Dict[str, Any]]:
    """
    Generate professional presentation content using CrewAI-style multi-agent approach
    """
    crew = PPTGeneratorCrew()
    return crew.generate_presentation(topic, requirements, num_slides)

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python crew_ai.py <topic> [requirements]")
        sys.exit(1)
    
    topic = sys.argv[1]
    requirements = sys.argv[2] if len(sys.argv) > 2 else ""
    
    print(f"🚀 Starting CrewAI PPT Generation for: {topic}")
    print(f"📋 Additional Requirements: {requirements or 'None'}")
    print("-" * 50)
    
    slides = generate_presentation_content(topic, requirements)
    
    print("\n" + "="*50)
    print("📄 GENERATED PRESENTATION SUMMARY")
    print("="*50)
    for slide in slides:
        print(f"Slide {slide['id'] + 1}: {slide['title']} ({slide['slideType']})")
        if 'agent_notes' in slide:
            print(f"  👥 {slide['agent_notes']}")
    
    print(f"\n🎯 Total Slides Generated: {len(slides)}")
    print(json.dumps(slides, indent=2))
